java/fury-core/src/main/java/org/apache/fury/Fury.java

```diff
@@ -117,6 +116,7 @@ public final class Fury implements BaseFury {
   private Iterator<MemoryBuffer> outOfBandBuffers;
   private boolean peerOutOfBandEnabled;
   private int depth;
+  private final MemoryBufferManager bufferManager;
```

Comment:
The buffer logic is pretty simple, I don't think it deserves a separate abstraction.  The abstraction is not always a good thing.

Reply:
Could we apply `execute` method in `Fury`?

Reply:
I prefer not, we should reduce callstack and avoid use lambda here to reduce performance cost. There are alreay several nested call before the real serialization logic run

Reply:
Okay, I got it, I will close this PR later.